lib/command.js

```diff
@@ -1096,12 +1096,16 @@ Expecting one of '${allowedValues.join("', '")}'`);
       this.help();
     }
     const subCommand = this._findCommand(subcommandName);
-    if (subCommand && !subCommand._executableHandler) {
+    if (!subCommand) {
+      this.help({ error: true });
```

Comment:
No, this is wrong. This routine is only called to implement the help command. It might be invoked by end-user with a subcommand name:
```
program help
program help subcommand
```

It might help to look at the PR, which was about invoking help on a subcommand more robust: https://github.com/tj/commander.js/pull/1864

Reply:
Oh, but might not be valid subcommand name.
Oh, but that is handled by fall through case.

In summary, this piece of code is about directly calling subcommand help if we can reach it, so don't need to worry about help flags or help command.

Reply:
What exactly is wrong?

If you mean the `this.help({ error: true })` call, that is exactly what currently happens in `._dispatchSubcommand()` if no subcommand with the supplied name is found.

Reply:
Right. Which I am happy happening there. This routine is short-circuiting one case without refactoring the existing somewhat complex code.